src/Common/tests/System/PlatformDetection.cs

```diff
@@ -207,49 +207,72 @@ private static bool IsDistroAndVersion(string distroId, string versionId = null)
         }
 
-        private static IdVersionPair ParseOsReleaseFile()
+        public static string GetDistroVersionString()
+        {
+            if (!RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return "";
+            }
+
+            DistroInfo v = ParseOsReleaseFile();
+
+            return "Distro=" + v.Id + " VersionId=" + v.VersionId + " Pretty=" + v.PrettyName + " Version=" + v.Version;
+        }
+
+        private static DistroInfo ParseOsReleaseFile()
         {
             Debug.Assert(RuntimeInformation.IsOSPlatform(OSPlatform.Linux));
 
-            IdVersionPair ret = new IdVersionPair();
+            DistroInfo ret = new DistroInfo();
             ret.Id = "";
             ret.VersionId = "";
+            ret.Version = "";
+            ret.PrettyName = "";
 
             if (File.Exists("/etc/os-release"))
             {
                 foreach (string line in File.ReadLines("/etc/os-release"))
                 {
                     if (line.StartsWith("ID=", System.StringComparison.Ordinal))
                     {
-                        ret.Id = line.Substring("ID=".Length);
+                        ret.Id = RemoveQuotes(line.Substring("ID=".Length));
                     }
                     else if (line.StartsWith("VERSION_ID=", System.StringComparison.Ordinal))
                     {
-                        ret.VersionId = line.Substring("VERSION_ID=".Length);
+                        ret.VersionId = RemoveQuotes(line.Substring("VERSION_ID=".Length));
+                    }
+                    else if (line.StartsWith("VERSION=", System.StringComparison.Ordinal))
+                    {
+                        ret.Version = RemoveQuotes(line.Substring("VERSION=".Length));
+                    }
+                    else if (line.StartsWith("PRETTY_NAME=", System.StringComparison.Ordinal))
+                    {
+                        ret.PrettyName = RemoveQuotes(line.Substring("PRETTY_NAME=".Length));
                     }
                 }
             }
 
-            string versionId = ret.VersionId;
-            if (versionId.Length >= 2 && versionId[0] == '"' && versionId[versionId.Length - 1] == '"')
-            {
-                // Remove quotes.
-                ret.VersionId = versionId.Substring(1, versionId.Length - 2);
-            }
+            return ret;
+        }
 
-            if (ret.Id.Length >= 2 && ret.Id[0] == '"' && ret.Id[ret.Id.Length - 1] == '"')
+        private static string RemoveQuotes(string s)
+        {
+            s = s.Trim();
+            if (s.Length >= 2 && s[0] == '"' && s[s.Length - 1] == '"')
             {
                 // Remove quotes.
-                ret.Id = ret.Id.Substring(1, ret.Id.Length - 2);
+                s = s.Substring(1, s.Length - 2);
             }
 
-            return ret;
+            return s;
         }
 
-        private struct IdVersionPair
+        private struct DistroInfo
```

Comment:
FYI - There is a request to add similar information as a public API in RuntimeInformation see https://github.com/dotnet/corefx/issues/16629.